c_reference/include/conv_utils.h

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+#ifndef __CONVLAYER_UTILS__
+#define __CONVLAYER_UTILS__
+
+/**
+ * @brief Definition for the Utility Fucntion for Preparing the Low Rank Conv Weights
+ * @param[out]   out     pointer to the output signal, size (regular) = out_T * in_channels * kernel_size, size (depthwise) = out_t * kernel_size
+ * @param[in]    W1      1st component of the low rank weight tensor. size = out_channels * rank
+ * @param[in]    W2      2nd component of the low rank weight tensor. size (regular) = rank * in_channels * kernel_size, size (depthwise)  = rank * kernel_size
+ * @param[in]    rank    rank of the weight tensor. Low Rank
+ * @param[in]    I       dim 0 for W1, value = out_channels
+ * @param[in]    J       dim 1 for W2, value = in_channels * kernel_size, (or for depthwise) = kernel_size
+ */
+int matmul(float* out, float* W1, float* W2, unsigned rank, unsigned I, unsigned J);
```

Comment:
why are matmil and tanhgate special to conv utils? why cant existing gates be reused?

Reply:
I couldn't find any existing implementation of the matmul operation in utils. Hence, I defined it another file to avoid merge confilcts. If recommended I can move this function to the utils.c and utils.h files

The tanhgate is a custom made activation function for our specific network. It operates different from the regular tanh activation. It uses both tanh and sigmoid and reduces channel sizes. It's mainly designed for our network. If recommended I can move this either to dscnn or to utils

Reply:
@harsha-simhadri , this custom tanh gate is specific to the architecture I am using for keyword spotting and is motivated from anthony's work and seemed to be really better compared to other standard gates

Reply:
I would suggest moving matmul to the file matvec is at today. It would be good if we can generalize the two into one operator. @ShikharJ what do you think?

Re. the tanh gate, could you please give it a more specific name to differentiate it from the normal tanh and move it to the file which has other gates?

Reply:
Renamed TanhGate to semi_sigmoid_tanh. I also moved both the functions  matmul and the new activation from conv_utils.c to utils.c. I've removed conv_utils as well

Reply:
> I would suggest moving matmul to the file matvec is at today. It would be good if we can generalize the two into one operator. @ShikharJ what do you think?

Merging the two operators would require a transpose of the second matrix in the `matmul` operation. I think it's fine to keep them separate, since this is floating point code anyway.